core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -1203,4 +1204,47 @@ default StreamMessage<T> subscribeOn(EventExecutor eventExecutor) {
         requireNonNull(eventExecutor, "eventExecutor");
         return new SubscribeOnStreamMessage<>(this, eventExecutor);
     }
+
+    /**
+     * Configures a timeout for the stream based on the specified duration.
+     * The default timeout mode is {@link StreamTimeoutMode#UNTIL_NEXT}.
+     *
+     * <p>Example usage:
+     * <pre>{@code
+     * StreamMessage<String> stream = ...;
+     * StreamMessage<String> timeoutStream = stream.timeout(Duration.ofSeconds(10));
+     * }</pre>
+     *
+     * @param timeoutDuration the duration before a timeout occurs
+     * @return a new {@link TimeoutStreamMessage} with the specified timeout duration and default mode
```

Comment:
`TimeoutStreamMessage` is an internal class which shouldn't be exposed in Javadoc.
```suggestion
     * @return a new {@link StreamMessage} with the specified timeout duration and default mode
```